### DISCUSSION TITLE
RFEM 6 GUI not visible (run on background

### DISCUSSION THREAD

**Discussion by Dervis-vL** [Dlubal-Software/RFEM_Python_Client#259] [Q&A] [answered]
Hi, 

I would like to boost a simple projects productivity by not having the RFEM 6 GUI open, but compute my calculations on the background.

Is there a way to do this like visible=False or headless=True?

Thank you!

**Comment by OndraMichal** [comment, marked as answer]
Hi @Dervis-vL,
I'm glad you asked. Indeed we have such solution. The name of it is [RFEM 6 Server](https://www.dlubal.com/en/solutions/online-services/webservice-and-api#:~:text=There%20is%20also%20an%20RFEM%C2%A06%20server%20that%20runs%20on%20your%20computer%20without%20a%C2%A0GUI%2C%20but%20only%20responds%20to%20your%20web%20service%20requests.). It is RFEM6 running in command line 🥇  It is part of a standard installation. 

As of today there are few changes in arguments. So with next release Monday June 5th 2023 (or Tuesday) you can start it with: 
`.\RFEM6Server.exe --email=jane.doe@gmail.com --password=125 --license=000005-25 --start-soap-server=8081 --soap-number-of-model-server-ports=10`

**Reply by gvandewiel86** [reply]
Hi @OndraMichal ,

I was wondering if there is more documentation available regarding the RFEM6 server with respect to command line arguments etc.
Furthermore is it only possible to run RFEM6 server with a webservices license or is it available for all RFEM users?

**Reply by Saeiiiid** [reply]
is there any documentation for the inputs for the licenses? I mean where the license number = 000005-25 comes from?

**Reply by dogukankaratas** [reply]
Hi @gvandewiel86,

RFEM6Server is installed automatically when RFEM6 installed in a local computer, so you don't need any other license. But the advantage of the RFEM6Server is that you can use Webservices commands without using any GUI (it's faster and applicable to other networks.) 

That's why I see no point to use RFEM6Server without any Webservices operations.

**Reply by dogukankaratas** [reply]
Hi @Saeiiiid,

You can find your license number inside RFEM6 --> Help --> System Information.

![image](https://github.com/Dlubal-Software/RFEM_Python_Client/assets/61163577/62165d61-6a50-41f5-a753-8f252b54cac5)

**Reply by Saeiiiid** [reply]
Thanks for your reply. still I get the error as below and I don't know what exactly is going wrong. 

2023/08/29 16:17:34 635638      INFO [commonMain.cpp->commonMain:64]    Initializing application...
2023/08/29 16:17:36 553069      INFO [DApplicationBase.cpp->DApplicationBase::registerBinaryResources:195]      Loading resources...
2023/08/29 16:17:36 604027      INFO [DApplicationBase.cpp->DApplicationBase::migrateSettings:162]      Migrating settings...
2023/08/29 16:17:36 623120      INFO [DApplicationBase.cpp->DApplicationBase::initApplicationConfigs:174]       Initializing application config...
2023/08/29 16:17:41 538800      FATAL [commonMain.cpp->commonMain:68]
        *******  EXIT trigger caused by LOG(FATAL) entry:
        "Failed to initialize application
******* STACKDUMP *******
stack dump [0]
stack dump [1]
stack dump [2]
stack dump [3]
stack dump [4]
stack dump [5]
stack dump [6]
Exiting after fatal event  (FATAL). Fatal type:  SIGABRT
Log content flushed sucessfully to sink

**Comment by Dervis-vL** [comment]
I have not seem to get the RFEMServer.exe running. I get a fatal error
`FATAL: Failed to parse program arguments: Unknown options: email, license, start-soap-server.`

Is there more documentation regarding the RFEMServer.exe?

**Reply by dogukankaratas** [reply]
Hi @Dervis-vL,

It could be an error wrong path selection. Please first go to the folder order where you have RFEM6Server.exe. Then run

`RFEM6Server.exe --email=your@email.com --password=your_password --license=your_license_number --start-soap-server=8081 --soap-number-of-model-server-ports=10`

Please let me know if you have further problems.

**Reply by Dervis-vL** [reply]
I previously tested running RFEM6Server.exe in the following folder that contains RFEM6Server.exe:
`C:\Program Files\Dlubal\RFEM 6.02\bin`

I also use the email as specified in the 'System Information' window. How could I check if I am using the correct password?

**Reply by Saeiiiid** [reply]
remove "--email=" and  "--password=" from the command.

**Reply by Dervis-vL** [reply]
I still have negative results where the arguments do not seem to exist. I am using RFEM 6.02. Could this be the issue?

**Reply by OndraMichal** [reply]
Hi,
I just checked and it works. There is one caveat though. If you forgot to put space between parameters (' --') it will generate _Failed to login to Dlubal account_ message.

**Reply by Dervis-vL** [reply]
@OndraMichal 
Thank you for your response.

With the folder `C:\Program Files\Dlubal\RFEM 6.02\bin` as my cwd and using the same credentials I use for my account on dlubal.com, I run the RFEM6Server.exe with arguments as you specified (copy pasted your response above, with whitespace between the args). 

This results in the fatal error:
`FATAL: Failed to parse program arguments: Unknown options: email, license, start-soap-server.`

Since there is no documentation, could you take us through a step by step proces, possibly with print screens, since we are unable to trace the issue (or difference between both of our proces).

**Reply by OndraMichal** [reply]
Is your version newer that June 5th 2023? The parameters changed with version 6.02.0068.

**Reply by Dervis-vL** [reply]
This will be the issue. Will download the newest version and try agin. Thanks

**Reply by gvandewiel86** [reply]
If there would be any published documentation regarding the rfem server perhaps we wouldn’t be running in these kinds of issues.

However based on the lack of response the multiple questions regarding any documentation it seems there is no documentation (publicly) available?

If parameters change and it is nowhere documented how do you expect users to keep up with the changes?

**Reply by OndraMichal** [reply]
Hi,
this is still under development, so there may be some changes that break backwards compatibility. But when we ran into such issue, we try to explain it in as much detail as possible. There is no official manual with pictures and step-by-step instructions, but the change in parameters was mentioned in the first comment here in the discussion. And if you search for a few seconds you will find wiki article called [RFEM6Server RFEM without GUI](https://github.com/Dlubal-Software/RFEM_Python_Client/wiki/RFEM6Server-RFEM-without-GUI) which carries all the information we have about it right now.

**Reply by Dervis-vL** [reply]
To clarify on this thread, what I have understood and tried, RFEM6Server.exe works from versions 6.02.0068 and up until 6.03.xxxx. RFEM6Server.exe does not work in v6.03.xxxx and is under development. Logging in with RFEM6 Server worked on v6.02.0074 for instance. 

I will closely look out for the development of rfem6 server for v6.03 and up, and hope to get updated when deployed.

**Reply by OndraMichal** [reply]
Hi @Dervis-vL,
I'm not sure where you got that "RFEM6Server.exe does not work in v6.03.xxxx". It works. It's just still under development and we cannot guarantee that there will be no parameter changes in the future.

**Reply by Dervis-vL** [reply]
My bad. I got this information from msc engineer.
